integ-test/src/test/java/org/opensearch/sql/ppl/ShowDataSourcesCommandIT.java

```diff
@@ -22,6 +23,18 @@ protected void init() throws Exception {
     loadIndex(Index.DATASOURCES);
   }
 
+  /**
+   * Integ tests are dependent on self generated metrics in prometheus instance.
+   * When running individual integ tests there
+   * is no time for generation of metrics in the test prometheus instance.
+   * This method gives prometheus time to generate metrics on itself.
+   * @throws InterruptedException
+   */
+  @BeforeClass
+  protected static void metricGenerationWait() throws InterruptedException {
+    Thread.sleep(20000);
```

Comment:
This will extend CI run for 2 min just for sleeping. Is it possible to do a conditional sleep?

Reply:
I need to query prometheus probably which is not feasible.
Probably I can reduce this to 10sec.

Reply:
It will be great if you can, or consider this to follow-up update:
```
while (!isMetricsGenerated()) {
  sleep(100);
}
```

Reply:
will follow up in the next release. Thanks.